.github/workflows/examples-java-codeql-analysis.yml

```diff
@@ -0,0 +1,43 @@
+name: "CodeQL Analysis"
+
+on:
+  push:
+  pull_request:
+    paths:
+      - "examples/java/**"
+      - ".github/workflows/examples-java-codeql-analysis.yml"
+  schedule:
+    - cron: '36 12 * * 0'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: examples/java
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language:
+          - 'java'
```

Comment:
actually, as this is just a Java-specific workflow, I'd say we can remove this and just inline it in the execution